testnet/stacks-node/src/run_loop/nakamoto.rs

```diff
@@ -218,13 +225,49 @@ impl RunLoop {
             .map(|e| (e.address.clone(), e.amount))
             .collect();
 
-        // TODO (nakamoto-neon): check if we're trying to setup a self-signing network
-        //  and set the right genesis data
+        let agg_pubkey_boot_callback = if let Some(self_signer) = self.config.self_signing() {
```

Comment:
Ah this was the key I was needing. Was wondering how to set boot contract state, but couldn't seem to figure it out!